Kernel/VM/MemoryManager.cpp

```diff
@@ -433,13 +433,8 @@ Region* MemoryManager::kernel_region_from_vaddr(VirtualAddress vaddr)
 
 Region* MemoryManager::user_region_from_vaddr(Space& space, VirtualAddress vaddr)
 {
-    // FIXME: Use a binary search tree (maybe red/black?) or some other more appropriate data structure!
     ScopedSpinLock lock(space.get_lock());
-    for (auto& region : space.regions()) {
-        if (region.contains(vaddr))
-            return &region;
-    }
-    return nullptr;
+    return space.find_region_containing({ vaddr, 1 });
```

Comment:
Can we do this for kernel_regions as well above? See: `MemoryManager::kernel_region_from_vaddr(VirtualAddress vaddr)`

Reply:
Kernel regions are stored as a simple linked list, so i cant run this method on them, unless youre suggesting to replace the kernel regions container itself?

Reply:
Yea I was suggesting we unify the implementation so they both use the same data structure. For example the page fault handler will benefit from the faster lookup.

Reply:
Hmm, yeah, i guess so, but it looks like memory manager sometimes just wants to iterate over all the kernel and user regions regardless of correct order, so that would be slowed down if we replace it with a red black tree, so we need to store both a red black tree and the current linked list (and use them for those 2 seperate cases), if thats fine

Reply:
Is there something particularly performance-sensitive that would benefit from unordered iteration? Otherwise I feel like we could go with simplicity first and then add on something like a linked list of regions as an optimization if this becomes an issue in the future.

Reply:
theres a FIXME in VMObject::for_each_region which says we unfortunately iterate through *all* regions (all user regions of all processes, and all kernel regions) every time an inode changes (but im not sure if thats still correct)

Reply:
Oh right, that one! I think we can improve that by finding a less generic way to locate all the `Region` objects that reference a given `InodeVMObject` :)

Perhaps `Region` could participate in an intrusive list of `InodeVMObject` clients, for example.

Reply:
that might be a bit out of scope for this PR (and for my current level of understanding of serenity's VM implementation :P), so would you prefer that i still switch over to an RB tree for kernel regions and just leave the current FIXME, or keep it a linked list for now and add a FIXME so the future PR that optimizes that unnecessary iteration can switch to an RB tree as well?